li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/DeliveredMessageOffsetTracker.java

```diff
@@ -192,11 +202,34 @@ private boolean hasDeliveredMessages(PartitionOffsetTracker offsetTracker) {
     }
 
     void updateCurrentSafeOffset(long currentSafeOffset) {
+      if (_delivered != -1) {
+        if (currentSafeOffset == _delivered) {
+          _partitionSnag = 0;
+        } else {
+          if (currentSafeOffset > _currentSafeOffset) { // safe offset moves forward
+            _partitionSnag -= (currentSafeOffset - _currentSafeOffset);
```

Comment:
cant we always say that snag = _delivered - _safeOffset  and just calculate it instead of trying to "diff" it?

Reply:
Sure we can do that. I think the complexity of the "Diff" is not needed for this coarse metric.